odbc/src/api/functions.rs

```diff
@@ -375,21 +409,57 @@ pub extern "C" fn SQLDriverConnect(
     _string_length_2: *mut SmallInt,
     _drive_completion: DriverConnectOption,
 ) -> SqlReturn {
+    dbg!();
     unsupported_function(MongoHandleRef::from(connection_handle), "SQLDriverConnect")
 }
 
 #[no_mangle]
 pub extern "C" fn SQLDriverConnectW(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _window_handle: HWnd,
-    _in_connection_string: *const WChar,
-    _string_length_1: SmallInt,
+    in_connection_string: *const WChar,
+    string_length_1: SmallInt,
     _out_connection_string: *mut WChar,
     _buffer_length: SmallInt,
     _string_length_2: *mut SmallInt,
     _driver_completion: DriverConnectOption,
 ) -> SqlReturn {
-    unimplemented!()
+    dbg!();
+    dbg!("!!!!");
+    let conn_handle = MongoHandleRef::from(connection_handle);
+    let conn = (*conn_handle).as_connection();
+    if conn.is_none() {
+        conn_handle.add_diag_info(ODBCError::InvalidHandleType("Connection"));
+        return SqlReturn::ERROR;
+    }
+    let conn = conn.unwrap();
+    let uri = input_wtext_to_string(in_connection_string, string_length_1 as usize);
+    let database = env::var("SQL_ATTR_CURRENT_CATALOG").ok();
+    let connect_result = mongo_odbc_core::MongoConnection::connect(
+        &uri,     // uri
+        database, // current_db
+        None,     // op timeout i32
+        None,     // login timeout i32
+    );
+    match connect_result {
+        Ok(mc) => {
+            let mut conn_writer = conn.write().unwrap();
+            conn_writer.attributes.current_db = mc.current_db.clone();
```

Comment:
[Just a comment]
I didn't find anything in the spec which says that we should set SQL_ATTR_CURRENT_CATALOG attribute outside of a SQLSetConnAttr call but I think it makes sense here.